lib/puppet/application.rb

```diff
@@ -281,6 +281,16 @@ def run_mode( mode_name = nil)
       @run_mode = Puppet::Util::RunMode[ mode_name || Puppet.settings.preferred_run_mode ]
     end
 
+    def environment_mode( mode_name = nil)
+      if mode_name
+        Puppet.settings.preferred_environment_mode = mode_name
```

Comment:
I don't see a reason to have the preferred_environment_mode infrastructure. I think we can just hard-code a default here.

Reply:
If possible, I'd like to see this setting reflected in Help text somehow.

Reply:
It probably just needs to be a note in the help text of individual applications that they don't require the environment directory to exist.